### PR TITLE
Organize imports in `sfv2`

### LINKED ISSUE #2501: Remove extra new lines from imports in `sfv2`

### PR DESCRIPTION
**Pull request by eureka-cpu**
This removes newlines from imports so they can be properly organized.
Closes #2501

### CODE REVIEW

**Review by JoshuaBatty — DISMISSED** [review]

File: sway-fmt-v2/src/config/comments.rs [hunk 1]
```diff
@@ -1,7 +1,6 @@
 //! Configuration options related to formatting comments.
-use crate::constants::DEFAULT_MAX_COMMENT_WIDTH;
-
 use super::user_opts::CommentsOptions;
+use crate::constants::DEFAULT_MAX_COMMENT_WIDTH;
```
Comment (kayagokalp) [inline sway-fmt-v2/src/config/comments.rs:3]: Do we want to keep this kind of separate imports rooted from `super::`? If we also convert those imports to `crates::` type (and add it to the others) they would also be ordered by `rustfmt`. But I would also understand if you prefer having these separated as it is a shorter relative path. 

```suggestion
use crate::{config::user_opts::CommentsOptions, constants::DEFAULT_MAX_COMMENT_WIDTH};
```

File: sway-fmt-v2/src/config/heuristics.rs [hunk 2]
```diff
@@ -1,13 +1,12 @@
 //! Configuration options related to heuristics.
+use super::user_opts::HeuristicsOptions;
```
Comment (kayagokalp) [inline sway-fmt-v2/src/config/heuristics.rs:2]: ^ Same

File: sway-fmt-v2/src/config/items.rs [hunk 3]
```diff
@@ -1,8 +1,7 @@
 //! Configuration options related to item formatting.
-use serde::{Deserialize, Serialize};
-
 use super::{lists::ListTactic, user_opts::ItemsOptions};
 use crate::constants::{DEFAULT_BLANK_LINES_LOWER_BOUND, DEFAULT_BLANK_LINES_UPPER_BOUND};
```
Comment (kayagokalp) [inline sway-fmt-v2/src/config/items.rs:3]: ^ Same

File: sway-fmt-v2/src/config/whitespace.rs [hunk 4]
```diff
@@ -1,9 +1,7 @@
 //! Standard system and editor whitespace configuration options. Advanced whitespace options will be deferred to their corresponding sub-classes.
-use serde::{Deserialize, Serialize};
-
-use crate::constants::{CARRIAGE_RETURN, DEFAULT_MAX_LINE_WIDTH, DEFAULT_TAB_SPACES, LINE_FEED};
-
 use super::user_opts::WhitespaceOptions;
+use crate::constants::{CARRIAGE_RETURN, DEFAULT_MAX_LINE_WIDTH, DEFAULT_TAB_SPACES, LINE_FEED};
```
Comment (kayagokalp) [inline sway-fmt-v2/src/config/whitespace.rs:3]: ^ Same

File: sway-fmt-v2/src/utils/attribute.rs [hunk 5]
```diff
@@ -1,3 +1,7 @@
+use super::{
+    bracket::{Parenthesis, SquareBracket},
+    comments::{ByteSpan, LeafSpans},
+};
```
Comment (kayagokalp) [inline sway-fmt-v2/src/utils/attribute.rs:4]: Same ^

File: sway-fmt-v2/src/utils/expr/mod.rs [hunk 6]
```diff
@@ -1,3 +1,7 @@
+use super::{
+    bracket::{CurlyBrace, Parenthesis, SquareBracket},
+    shape::{ExprKind, LineStyle},
+};
```
Comment (kayagokalp) [inline sway-fmt-v2/src/utils/expr/mod.rs:4]: ^

File: sway-fmt-v2/src/utils/generics.rs [hunk 7]
```diff
@@ -1,9 +1,8 @@
 use super::bracket::{close_angle_bracket, open_angle_bracket};
+use super::shape::LineStyle;
```
Comment (kayagokalp) [inline sway-fmt-v2/src/utils/generics.rs:2]: ^

File: sway-fmt-v2/src/utils/path.rs [hunk 8]
```diff
@@ -1,3 +1,4 @@
+use super::bracket::{close_angle_bracket, open_angle_bracket};
```
Comment (kayagokalp) [inline sway-fmt-v2/src/utils/path.rs:1]: ^

File: sway-fmt-v2/src/utils/pattern.rs [hunk 9]
```diff
@@ -1,3 +1,4 @@
+use super::bracket::{CurlyBrace, Parenthesis};
```
Comment (kayagokalp) [inline sway-fmt-v2/src/utils/pattern.rs:1]: ^

File: sway-fmt-v2/src/utils/program_type.rs [hunk 10]
```diff
@@ -1,11 +1,9 @@
+use super::comments::{ByteSpan, LeafSpans};
+use crate::FormatterError;
```
Comment (kayagokalp) [inline sway-fmt-v2/src/utils/program_type.rs:2]: ^

File: sway-fmt-v2/src/utils/punctuated.rs [hunk 11]
```diff
@@ -1,3 +1,4 @@
+use super::shape::LineStyle;
```
Comment (kayagokalp) [inline sway-fmt-v2/src/utils/punctuated.rs:1]: ^

File: sway-fmt-v2/src/utils/ty.rs [hunk 12]
```diff
@@ -1,3 +1,4 @@
+use super::shape::LineStyle;
```
Comment (kayagokalp) [inline sway-fmt-v2/src/utils/ty.rs:1]: ^

File: sway-fmt-v2/src/utils/where_clause.rs [hunk 13]
```diff
@@ -1,10 +1,9 @@
+use super::comments::{ByteSpan, LeafSpans};
```
Comment (kayagokalp) [inline sway-fmt-v2/src/utils/where_clause.rs:1]: ^

**Review by kayagokalp — COMMENTED** [review]
Looking good! I just left a single nit (on multiple places) about mixing `super::` and `crate::` type paths. I think if we also merge those automatic `rustfmt` ordering that we are trying to ensure with this PR will benefit a little more but I understand that addressing the same piece by `crate::` rather than `super::` will be longer, so I guess this is more like a design decision and thus I am leaving this up to you. I just wanted to provide an option to consider 👍 Since automatic merge is enabled for this PR, I am not approving right now to provide a chance for you to look at the review but if you decide to go like this, you can ping me and I will happily approve.

File: sway-fmt-v2/src/config/comments.rs (hunk 1 above)
Comment (eureka-cpu) [reply sway-fmt-v2/src/config/comments.rs:3]: Personally I like `super` (: if @mitchmindtree or @JoshuaBatty prefer `crate` only then I'd gladly change it. But I have no qualms as is!

Comment (mitchmindtree) [reply sway-fmt-v2/src/config/comments.rs:3]: I think `crate` is perhaps a little more conventional, though `super` can be useful when you're in a module that you know will always be nested and have the same relative path, e.g.

```rust
// Inside module `foo.rs`

fn foo() { ... }

mod tests {
    use super::foo;
    // ...
}
```

^ Even if `foo.rs` gets moved or nested within another module, `tests` will always be nested within `foo.rs` so it's pretty safe to use `super` from within `tests`.

For most of the cases in this lib I'd personally go with `crate`.

Comment (JoshuaBatty) [reply sway-fmt-v2/src/config/comments.rs:3]: Yeah I prefer `crate`, makes it a bit more flexible for the reasons mitchie stated above.

Comment (eureka-cpu) [reply sway-fmt-v2/src/config/comments.rs:3]: gotcha, I'll add those changes in the morning

File: sway-fmt-v2/src/items/item_use.rs [hunk 14]
```diff
@@ -7,6 +7,7 @@ use crate::{
     },
 };
 use std::fmt::Write;
+use std::vec;
```
Comment (kayagokalp) [inline sway-fmt-v2/src/items/item_use.rs:10]: Maybe we can also merge them like:

```suggestion
use std::{fmt::Write, vec};
```

**Review by kayagokalp — COMMENTED** [review]
Just left a single nit that might be slipped through otherwise I think it is good to go!

File: sway-fmt-v2/src/items/item_use.rs (hunk 14 above)
Comment (kayagokalp) [reply sway-fmt-v2/src/items/item_use.rs:10]: Also I think there are no problems if we do not have this `use` for `vec`, if you prefer we can also drop it.

**Review by kayagokalp — APPROVED** [review]

**Review by JoshuaBatty — APPROVED** [review]

**Pull request merged**